### PR TITLE
fix(spaces): update delete spaces api to support skipCluster argument

### PR DESCRIPTION
**Pull request by jiekang**
Title says it all. This is for https://github.com/openshiftio/openshift.io/issues/3500#issuecomment-406383100

Tests are added and some of the code is cleaned up.

### CODE REVIEW

**Comment by jiekang** [comment]
@ebaron yes.I was planning to open one after this was merged (different repositories).

**Comment by ebaron** [comment]
@jiekang I see that now. Thanks!

File: src/app/spaces/space.service.ts [hunk 1]
```diff
@@ -104,8 +104,8 @@ export class SpaceService {
       .catch((error: any): Observable<Space> => this.handleError(error));
   }
 
-  deleteSpace(space: Space): Observable<Space> {
-    const url: string = `${this.spacesUrl}/${space.id}`;
+  deleteSpace(space: Space, skipCluster: boolean = false): Observable<Space> {
+    const url: string = `${this.spacesUrl}/${space.id}?skipCluster=${skipCluster}`;
```
Comment (andrewazores) [inline src/app/spaces/space.service.ts]: I'd rather use `URLSearchParams` here, rather than manually constructing the query string

Comment (jiekang) [reply src/app/spaces/space.service.ts]: Sounds good. It should be done now.

File: src/app/spaces/space.service.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { Observable } from 'rxjs';
 
 import { WIT_API_URL } from '../api/wit-api';
 import { Space } from '../models/space';
+import { query } from '@angular/core/src/animation/dsl';
```
Comment (andrewazores) [inline src/app/spaces/space.service.ts]: not needed?

Comment (jiekang) [reply src/app/spaces/space.service.ts]: Argh auto import on writing text. Sorry and thanks for the catch. Should be gone now.

**Pull request merged**